zebra/zebra_vxlan.c

```diff
@@ -7836,6 +7836,52 @@ void zebra_vxlan_remote_macip_add(ZAPI_HANDLER_ARGS)
 	return;
 }
 
+/*
+ * Handle remote vtep delete by kernel; re-add the vtep if we have it
+ */
+int zebra_vxlan_check_readd_vtep(struct interface *ifp,
+				 struct in_addr vtep_ip)
+{
+	struct zebra_if *zif;
+	struct zebra_vrf *zvrf = NULL;
+	struct zebra_l2info_vxlan *vxl;
+	vni_t vni;
+	zebra_vni_t *zvni = NULL;
+	zebra_vtep_t *zvtep = NULL;
+
+	zif = ifp->info;
+	assert(zif);
+	vxl = &zif->l2info.vxl;
+	vni = vxl->vni;
+
+	/* If EVPN is not enabled, nothing to do. */
+	if (!is_evpn_enabled())
+		return 0;
+
+	/* Locate VRF corresponding to interface. */
+	zvrf = vrf_info_lookup(ifp->vrf_id);
+	if (!zvrf)
+		return -1;
+
+	/* Locate hash entry; it is expected to exist. */
+	zvni = zvni_lookup(vni);
+	if (!zvni)
```

Comment:
shouldn't this return -1 ?

Reply:
function calls out `re-add it if we have it`.  Not finding the zvni seems like it's a ok state to be in and not considered a failure here imo.